akka-actor-tests/src/test/scala/akka/io/TcpConnectionSpec.scala

```diff
@@ -915,9 +923,11 @@ class TcpConnectionSpec extends AkkaSpec("""
       new ChannelRegistration {
         def enableInterest(op: Int): Unit = interestCallReceiver.ref ! op
         def disableInterest(op: Int): Unit = interestCallReceiver.ref ! -op
-        def cancel(): Unit = ()
+        def cancelAndClose(andThen: () ⇒ Unit): Unit = onCancelAndClose(andThen)
```

Comment:
Yep, we don't use the real SelectorHandler impl here, so we need to replicated the correct behavior.